src/components/Dropdown/Dropdown.tsx

```diff
@@ -308,6 +352,96 @@ const Dropdown = ({
 
   const tagContainerItemProps = valueItemTagProps.containerProps || {};
 
+  // effect to determine if user is scrolling up or down
+  useEffect(() => {
+    const threshold = 0;
+    let lastScrollY = window.pageYOffset;
+    let ticking = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        ticking = false;
+        return;
+      }
+      setIsScrollingDown(scrollY > lastScrollY);
+      lastScrollY = scrollY > 0 ? scrollY : 0;
+      ticking = false;
+    };
+
+    const onScroll = () => {
+      if (!ticking) {
+        window.requestAnimationFrame(updateScrollDir);
+        ticking = true;
+      }
+    };
+    window.addEventListener('scroll', onScroll);
+
+    return () => window.removeEventListener('scroll', onScroll);
+  }, []);
+
+  const buildThresholdArray = () => Array.from(Array(100).keys(), i => i / 100);
+  const intersectOptions = useMemo(() => {
+    return {
+      root: null,
+      rootMargin: '0px',
+      threshold: buildThresholdArray(),
+    };
+  }, []);
+
+  const intersectionCallback = useCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      if (isAware) {
+        if (entries.length === 1) {
+          const [entry] = entries;
+          // swap the dropdown to open downward if its hitting the top
+          if (
+            entry.intersectionRatio < 0.95 &&
+            entry.target === optionsContainerInternalRef.current
+          ) {
+            if (isScrollingDown) {
+              if (!isOpenedBelow && entry.intersectionRatio < prevIntersectionRatio) {
+                setIsOpenedBelow(true);
+              }
+            }
+          }
+          setPrevIntersectionRatio(entry.intersectionRatio);
```

Comment:
Could this be a ref instead of state? Reasons it could need to be a state var is if the rerender is needed for the app to react to intersection changes. Reason for it to be a ref is that it probably is just used for comparisons when the intersection changes and causes a re-render anyway.

My concern is that this may cause unnecessary rerenders or double renders